### PR TITLE
[urtrace] add json output support

### PR DESCRIPTION
**Pull request by pbalcer**
This is to match a similar functionality in `sycl-prof` that's based on PI.

left-heavy
urtrace-sandwich

### CODE REVIEW

File: tools/urtrace/collector.cpp [hunk 1]
```diff
@@ -149,27 +166,124 @@ static class cli_args {
...
-using namespace ur_params;
-
 typedef std::chrono::high_resolution_clock Clock;
+typedef std::chrono::time_point<Clock> Timepoint;
+
+class TraceWriter {
+  public:
+    virtual ~TraceWriter() {}
+    virtual void begin(uint64_t id, const char *fname, std::string args,
+                       Timepoint tp) = 0;
```
[the remark this answers is not in the document]
Comment (pbalcer) [reply tools/urtrace/collector.cpp]: Done.

File: tools/urtrace/collector.cpp [hunk 2]
```diff
@@ -149,27 +166,124 @@ static class cli_args {
...
     bool no_args;
+    enum output_format output_format;
     std::optional<std::string>
         filter_str; //the filter_str is kept primarly for printing.
     std::optional<std::regex> filter;
 } cli_args;
 
-using namespace ur_params;
-
 typedef std::chrono::high_resolution_clock Clock;
```
[the remark this answers is not in the document]
Comment (pbalcer) [reply tools/urtrace/collector.cpp]: Good point, thanks. Done.

**Review by PatKamin — APPROVED** [review]

**Review by kswiecicki — APPROVED** [review]

File: tools/urtrace/collector.cpp [hunk 3]
```diff
@@ -149,27 +166,121 @@ static class cli_args {
...
+    }
+};
+
+std::unique_ptr<TraceWriter> create_writer() {
+    switch (cli_args.output_format) {
+    case OUTPUT_HUMAN_READABLE:
+        return std::make_unique<HumanReadable>();
+    case OUTPUT_JSON:
+        return std::make_unique<JsonWriter>();
+        break;
```
Comment (DamianDuy) [inline tools/urtrace/collector.cpp:266]: This ```break``` here doesn't seem necessary.

Comment (pbalcer) [reply tools/urtrace/collector.cpp]: Thanks. Done.

**Review by kswiecicki — APPROVED** [review]

**Pull request merged**